### ISSUE TITLE
place for XCP-ng Center to host update.xml and release files

### ISSUE THREAD

**Issue by borzel** [xcp-ng/xcp#36]
@olivierlambert can you please
* create a place for XCP-ng Center to host it's update.xml and other release files?
or
* let a subdomain like center.xcp-ng.org point to my server?
or
* get in touch with me to find a good solution

Background: I dont want to implement the updatesystem for XCP-ng Center based on github. For privacy and other reasons.

**Comment by stormi** [comment]
Hi! Do you still need such a place? I think we could upload files to a subdirectory of https://updates.xcp-ng.org/ if @olivierlambert agrees.

**Comment by olivierlambert** [comment]
Because our update system is vastly different than Citrix, I don't see that's still relevant. XCP-ng Center should rely on our XAPI update plugin.

**Comment by stormi** [comment]
I thought that was for updating XCP-ng Center, not XCP-ng itself.

**Comment by olivierlambert** [comment]
Oh okay I thought it was for updating XCP-ng hosts. Why not having directly checking the XML on GitHub? This would be easier for you :)

**Comment by borzel** [comment]
read in my initial text:
> Background: I dont want to implement the updatesystem for XCP-ng Center (permanently) based on github. For privacy and other reasons.

**Comment by olivierlambert** [comment]
I'm not sure to understand? It's just a XML with numbers right? I probably don't get it, can you develop? The great thing about getting it on GitHub is there is no need to synchronize anything else than merging/pushing to modify the file and that's it.

I mean I'm not against hosting it on xcp-ng.org, but this will require extra work to have something sync with this repo anyway.

**Comment by borzel** [comment]
Of course its easier. Currently that's the way it works. Not a big problem at all.

The only downside is that github can now "see" how many connections are made to that xml file. Every XCP-ng host that checks for update connects to this file on a regular basis. Users can turn this check off, but I thought we can provide an alternative to it, so that Github is not collecting metadata from our updatingsystem.

I could also host this file on my own webserver, but an update url with xcp-ng.org in it would be better :-)

There is no need to have this file in Github.

**Comment by olivierlambert** [comment]
No, update requests for XCP-ng Center are **only coming from your XCP-ng Center client**. Not the host. So it's only your current Windows OS.

XCP-ng updates rely on the repository file content inside `/etc/yum.repos.d/xcp-ng.repo`, which today target `updates.xcp-ng.org` domain.

**Comment by borzel** [comment]
your are right, I meant XCP-ng Center Host... not the Server itself... was a typo

**Comment by borzel** [comment]
To sum up what was my goal: to geht a place where I can "upload" the xml file for updating purposes to keep the url's used in this project to one main domain and not relying on github for that.

Reasons:
* Github should not be able to collect metadata from our users
* keep the updating url clean

I understand that it is the easy way to keep the file on Github (because of that I do this currently).

I just wanted to have this discussed/overthought/guided. Maybe there is another option? Turn off the updating thing completely in XCP-ng Center? Create a subdomain like center.xcp-ng.org which I can controll/fill with data on my own?

**Comment by olivierlambert** [comment]
I can try to have a CRON-like script that fetch the files from GitHub every day so when a new version is available, it will be automatically copied on XCP-ng URL.

**Comment by borzel** [comment]
cron would be fine :-)

good that we are able to clear our communication problems with the help of normal human interaction ;-)

**Comment by olivierlambert** [comment]
Haha yes, we are totally humans behind screen, typing with our human hands

P.S: we are totally not :robot:

**Comment by borzel** [comment]
Rrrrrrobotttt ;-)
